qa/rpc-tests/mempool_size_limit.py

```diff
@@ -66,60 +73,128 @@ def load_from_storage(path):
-    txscript = create_tx_script()
+'''
+TODO: replace the default ones with the _noraw variants of the functions below
+def create_tx_script_noraw(script_size, amount):
+    bytes_per_out = script_size // NUM_OUT
+
+    random_bytes = hex_str_to_bytes("01" * bytes_per_out)
+
+    null_hash = hex_str_to_bytes("0000000000000000000000000000000000000000000000000000000000000000")
+    script_pubkey = CScript([OP_RETURN, OP_PUSHDATA2, bytes_per_out, random_bytes, OP_PUSHDATA1, 32, null_hash, OP_1, OP_CHECKBLOCKATHEIGHT])
+    print(script_pubkey)
+
+    return script_pubkey
+
+def create_big_transaction_noraw(node, tmpdir, input_t, script_size, fee, i):
+    assert(fee < input_t['amount'])
+    txscript = create_tx_script2(script_size, input_t['amount'] - fee)
+
+    inputs = []
+    inputs.append({ "txid" : input_t["txid"], "vout" : input_t["vout"], "scriptPubKey": input_t["scriptPubKey"]})
+    outputs = {node.getnewaddress(): Decimal((input_t['amount'] - fee)/NUM_OUT) for i in range(NUM_OUT)}
+    rawtx = node.createrawtransaction(inputs, outputs)
+
+    # build an object from the raw Tx in order to be able to modify it
+    tx_01 = CTransaction()
+    f = BytesIO(unhexlify(rawtx))
+    tx_01.deserialize(f)
+
+    for vout_idx in range(len(tx_01.vout)):
+        tx_01.vout[vout_idx].scriptPubKey = txscript
+
+    tx_01.rehash()
+    signresult = node.signrawtransaction(ToHex(tx_01), inputs, None, "NONE")
+    tsize = len(signresult['hex']) // 2
+    feerate = fee / tsize
+    write_to_file(tmpdir + "/tx" + str(i), feerate, signresult["hex"])
+    tx = {'feerate': feerate, 'hex': signresult["hex"]}
+    return tx, tsize
+'''
+
+def create_big_transaction(node, tmpdir, input_t, script_size, fee, i):
+    assert(fee < input_t['amount'])
+    txscript = create_tx_script(script_size, input_t['amount'] - fee)
+
+    inputs = []
+    inputs.append({ "txid" : input_t["txid"], "vout" : input_t["vout"], "scriptPubKey": input_t["scriptPubKey"]})
     addr = node.getnewaddress()
+    outputs = {addr: Decimal(0)} # This will be overwritten anyway
+    rawtx = node.createrawtransaction(inputs, outputs)
+    out_num_pos = rawtx.find('ffffffff')
+    script_position = rawtx.find('76a914')
+    script_length = int(rawtx[script_position - 2 : script_position], 16) * 2
+
+    newtx = rawtx[0:out_num_pos + 8]
+    newtx = newtx + txscript
+    newtx = newtx + rawtx[script_position + script_length:]
+
+    signresult = node.signrawtransaction(newtx, inputs, None, "NONE")
+    tsize = len(signresult['hex']) // 2
+    feerate = fee / tsize
+    if (tmpdir is not None):
+        write_to_file(tmpdir + "/tx" + str(i), feerate, signresult["hex"])
+    tx = {'feerate': feerate, 'hex': signresult["hex"]}
+    return tx
+
+
+def create_big_transactions(node, tmpdir, utxos, size):
     txs = []
     tot_size = 0
     i = 0
     os.makedirs(tmpdir + "/txs")
+    minfeerate, maxfeerate = MAX_FEE, Decimal("0")
+    fee = Decimal(0.5) + Decimal(0.01) * i
```

Comment:
Just a minor thing, as this might not impact the test at all, but to consider for future tests.
From `Decimal` module documentation:
`Decimal instances can be constructed from integers, strings, floats, or tuples. Construction from an integer or a float performs an exact conversion of the value of that integer or float.`
It implies that `Decimal(0.01)` is actually converted to `Decimal('0.01000000000000000020816681711721685132943093776702880859375')`, which in the long run may lead to wrong results in case of comparisons.
If we want to avoid the conversion, I suggest to use the constructor that takes a string as input parameter, i.e. `Decimal('0.01')`

Reply:
Great point! Applied your suggestion to all the occurrences that I found in the tests touched by this PR.

Reply:
thanks!